tests/client/ui/components/TestLoadingIndicator.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Simulate, renderIntoDocument } from 'react-addons-test-utils';
+
+import LoadingIndicator from 'ui/components/LoadingIndicator';
+
+
+describe('<LoadingIndicator />', () => {
+  function render(props) {
+    return renderIntoDocument(<LoadingIndicator {...props} />);
+  }
+
+  it('resets the animation after it ends', (done) => {
+    const root = render({ timeToReset: 5 });
+    const iconNode = ReactDOM.findDOMNode(root.loadingIcon);
+
+    assert.isTrue(root.state.animating);
+
+    Simulate.animationEnd(iconNode);
+
+    assert.isFalse(root.state.animating);
+
+    setTimeout(() => {
+      assert.isTrue(root.state.animating);
+      done();
+    }, 10);
```

Comment:
These kind of tests are hard to write but doing it like this may fail occasionally since the timing is dependent on how fast the Travis VM is processing events. Also, slowing down our test suite by 10 milliseconds should be avoided when possible.

I think you have a few options:
1. Figure out how to return a promise from `restartAnimation()` and call it directly. You could add a separate test to make sure the `animationEnd` event is hooked up.
2. Use mock injection to replace `setTimeout` altogether. You could do it by passing it as a prop, for example. With this approach you'd have to trigger the callback manually, like `mockSetTimeout.firstCall.args[0]()` to simulate `setTimeout` usage.
3. Maybe if you move it all to CSS you can delete these tests entirely :)